tests/conftest.py

```diff
@@ -0,0 +1,47 @@
+import attr
+from frozenlist2 import frozenlist
+
+
+def frozenlist_or_none_converter(obj, map_fn=(lambda x: x)):
+    if obj is not None:
+        return frozenlist(map_fn(obj))
+    return None
+
+
+@attr.s(kw_only=True, frozen=True)
+class FakePublishOptions(object):
```

Comment:
I didn't quite get the purpose of this.  The class in pulplib is public API, why not just import it and use it?

It could make sense if you want to avoid a direct dependency on pubtools-pulplib, though in that case you could make this class a lot more minimal - there's no need to reproduce all the doc strings and even the fields other than rsync_extra_args could be left out.

Reply:
Yeah, as this would've required waiting for https://github.com/release-engineering/pubtools-pulplib/pull/156 and a release of that lib and we really only need this piece of it, eliminating the dependency seems best, IMO.

Reply:
Ah, right.  Yeah, the dependency on an outstanding PR is annoying.

I think it is totally fine to use a custom object like this to avoid such dependencies, but if so it can be stripped down to the minimum possible, no need to copy an entire class.  It helps keep the tests clearer as well if it only has the fields it needs.